Source/Core/CesiumTerrainProvider.js

```diff
@@ -633,9 +659,11 @@ import TileProviderError from './TileProviderError.js';
             return undefined;
         }
 
-        var yTiles = provider._tilingScheme.getNumberOfYTilesAtLevel(level);
-
-        var tmsY = (yTiles - y - 1);
+        // The TileMapService scheme counts from the bottom left
+        if (!provider._scheme || provider._scheme === 'tms') {
+          var yTiles = provider._tilingScheme.getNumberOfYTilesAtLevel(level);
+          y = (yTiles - y - 1);
```

Comment:
This is the cause of the bug. Previously the y coordinate as left unchanged in the TMS case, so `createQuantizedMeshTerrainData` was given _Cesium_ coordinates (0=North) rather than TMS coordinates (0=South). After your change, `createQuantizedMeshTerrainData` is given TMS coordinates for TMS terrain, causing things to break.

Reply:
Ah I see, that makes sense. I've pushed a fix to the branch.